### PR TITLE
[Hexagon] Remove timeout on HAP_compute_res_acquire

### PR DESCRIPTION
**Pull request by Lunderberg**
When run on the simulator, a non-zero timeout for this call will busy-loop.

### CODE REVIEW

**Comment by Lunderberg** [comment]
@adstraw Non-urgent draft PR, for when you're back around.  I have a few other fixes I want to make to avoid triggering this issue, but having an error message is preferable to hanging indefinitely in case we hit it again in the future.  But you've also tested this more on physical hardware, where the non-zero timeout may be useful.

**Comment by csullivan** [comment]
cc @supersat

**Comment by supersat** [comment]
I noticed this same issue in HAP_compute_res_acquire when running a non-TVM test app under hexagon-sim. It's possible we could get this working under simulation (there's quite a bit of unexplained magic here: https://github.com/apache/tvm/blob/5dc40158e91fba255b791f4f381e1e68821225b9/src/runtime/hexagon/rpc/simulator/session.cc#L557-L565), but I imagine we'll need to revisit this code later anyway so immediately failing seems fine for now.

**Comment by kparzysz-quic** [comment]
> I noticed this same issue in HAP_compute_res_acquire when running a non-TVM test app under hexagon-sim.

Did you try to run it via `runelf.pbn` (i.e. via QuRT in a system emulation)?  In application emulation, `hexagon-sim` by itself only has a very basic set of OS services built into it, and the HAP functions are not a part of it.

The unexplained magic is setting up two cosims (additional subsystems to simulate): qtimer (a 19.2MHz clock) and l2vic (interrupt controller).  The parameters there are the hardware configuration of these subsystems and are pretty much hardwired.

**Comment by adstraw** [comment]
> @adstraw Non-urgent draft PR, for when you're back around. I have a few other fixes I want to make to avoid triggering this issue, but having an error message is preferable to hanging indefinitely in case we hit it again in the future. But you've also tested this more on physical hardware, where the non-zero timeout may be useful.

I can confirm that I see a hang running on device with a non-zero timeout as well.  We may want to root cause the hang, but suggest we move forward with this change as a workaround in the interim.

File: src/runtime/hexagon/hexagon/hexagon_buffer.cc [hunk 1]
```diff
@@ -84,7 +84,7 @@ struct VTCMAllocation : public Allocation {
     // allocate nbytes of vtcm on a single page
     HEXAGON_SAFE_CALL(HAP_compute_res_attr_set_vtcm_param(&res_info, /*vtcm_size = */ nbytes,
                                                           /*b_single_page = */ 1));
-    context_id_ = HAP_compute_res_acquire(&res_info, /*timeout = */ 10000);
+    context_id_ = HAP_compute_res_acquire(&res_info, /*timeout = */ 0);
```
[the remark this answers is not in the document]
Comment (Lunderberg) [reply src/runtime/hexagon/hexagon/hexagon_buffer.cc:90]: Good idea, and updated.

**Review by csullivan — APPROVED** [review]
